tensorpipe/test/channel/channel_test.h

```diff
@@ -203,39 +203,51 @@ class ChannelTestCase {
 
 template <typename TBuffer>
 class ClientServerChannelTestCase : public ChannelTestCase<TBuffer> {
-  std::vector<std::future<std::shared_ptr<tensorpipe::transport::Connection>>>
-  accept(
-      std::shared_ptr<tensorpipe::transport::Listener> listener,
-      size_t numConnections) {
-    auto promises = std::make_shared<std::vector<
-        std::promise<std::shared_ptr<tensorpipe::transport::Connection>>>>(
+  std::future<std::pair<
+      tensorpipe::Error,
+      std::vector<std::shared_ptr<tensorpipe::transport::Connection>>>>
```

Comment:
Nit: A type spanning three lines starts feeling like something we could address. Especially when one of these types is std::pair/tuple, which is often a shortcut for what could be a custom struct. So this could become `std::future<MultiAcceptResult>`, with `MultiAcceptResult` being a struct with two fields, `error` and `connections`.